crates/ntx-builder/src/builder/mod.rs

```diff
@@ -43,10 +46,39 @@ pub struct NetworkTransactionBuilder {
 }
 
 impl NetworkTransactionBuilder {
+    /// Creates a new instance of the network transaction builder.
+    pub fn new(
+        store_url: Url,
+        block_producer_url: Url,
+        rpc_url: Url,
+        tx_prover_url: Option<Url>,
+        ticker_interval: Duration,
+        bp_checkpoint: Arc<Barrier>,
+    ) -> Self {
+        Self {
+            store_url,
+            block_producer_url,
+            rpc_url,
+            tx_prover_url,
+            ticker_interval,
+            bp_checkpoint,
+        }
+    }
+
     pub async fn serve_new(self) -> anyhow::Result<()> {
         let store = StoreClient::new(self.store_url);
         let block_producer = BlockProducerClient::new(self.block_producer_url);
 
+        let rpc_client = ClientBuilder::new(self.rpc_url)
+            .with_tls()
+            .context("Failed to configure TLS")?
+            .with_timeout(Duration::from_secs(10))
+            .without_metadata_version()
+            .without_metadata_genesis()
+            .connect::<Rpc>()
+            .await
+            .context("Failed to connect to RPC server")?;
```

Comment:
As I mentioned above, I don't think this is necessary - we should be able to get note scripts directly from the store (may need to update protobuf files for this).